gateway-service/src/main/java/org/zowe/apiml/gateway/ribbon/RequestContextUtils.java

```diff
@@ -40,8 +40,12 @@ public static void setInstanceInfo(InstanceInfo info) {
 
     public static void addDebugInfo(String debug) {
         String existingDebugInfo = getDebugInfo();
-        RequestContext.getCurrentContext().set(DEBUG_INFO_KEY,
-            existingDebugInfo.isEmpty() ? debug : existingDebugInfo + "|" + debug);
+        if (existingDebugInfo.isEmpty()) {
+            RequestContext.getCurrentContext().set(DEBUG_INFO_KEY, debug);
+        } else if (!existingDebugInfo.contains(debug)) {
+            RequestContext.getCurrentContext().set(DEBUG_INFO_KEY, existingDebugInfo + "|" + debug);
+        }
```

Comment:
From the discussion in GH issue, it seems like the problem is somewhere else.

Hi @jandadav thank you for the meeting this morning.
We presented this topic at our team meeting and both considering the fact that this feature has already gone GA and that the team sees value in showing routing details as part of the request information that the Audit Log collects, we think that it should be kept. In this sense, we'd need to confirm if it is in fact a bug in how this debug information is being collected and appended.

I would say that extenders would like to keep track of all retries, even if they are on the same instance. 

The issue described by David in the last comment is only about the last "dangling" entry.